libs/blocks/chart/chart.js

```diff
@@ -487,22 +511,47 @@ export const getResponsiveSize = (authoredSize) => {
   return size;
 };
 
-/* c8 ignore next 19 */
+export const getLabelDegree = (chartStyles, isDesktop) => {
+  let diagonal = false;
+
+  if (isDesktop) {
+    diagonal = [...chartStyles]?.includes('desktop-diagonal-labels');
+  } else {
+    diagonal = [...chartStyles]?.includes('mobile-diagonal-labels');
+  }
+
+  return diagonal ? 60 : 0;
+};
+
+/* c8 ignore next 31 */
 const handleResize = (el, authoredSize, chartType, data, colors) => {
   const currentSize = getResponsiveSize(authoredSize);
   const previousSize = el?.getAttribute('data-responsive-size');
-  const previousIsLarge = previousSize === LARGE;
-  const currentIsLarge = currentSize === LARGE;
   const chartWrapper = el?.querySelector('.chart-wrapper');
   const chartInstance = window.echarts?.getInstanceByDom(chartWrapper);
+  const previousDevice = el?.getAttribute('data-device');
+  const width = window.innerWidth;
+  let currentDevice = 'mobile';
+  let isDesktop = false;
+
+  if (width >= DESKTOP_BREAKPOINT) {
+    currentDevice = 'desktop';
+    isDesktop = true;
+  }
 
   if (currentSize !== previousSize) {
     el.setAttribute('data-responsive-size', currentSize);
   }
 
-  if (previousIsLarge !== currentIsLarge) {
+  if (currentDevice !== previousDevice) {
```

Comment:
Now small and medium authored charts will also re-initialize when switching at `DESKTOP_BREAKPOINT`. I don't see a problem with that, just wanted to point it out.

Reply:
Yes, this is intentional. It allows the small and medium to have a different label orientation on desktop vs non-desktop